pom.xml

```diff
@@ -451,5 +454,22 @@ Note: The maven build currently needs to be started from the javaslang root dir
                 </pluginManagement>
             </build>
         </profile>
+        <!-- A profile for running the benchmarks -->
+        <profile>
+            <id>benchmark</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <version>${maven.exec.version}</version>
+                        <configuration>
+                            <classpathScope>test</classpathScope>
+                            <mainClass>javaslang.JmhRunner</mainClass>
```

Comment:
@talios Great! I fiddled around now a bit. This is my current profile in the parent pom:

```xml
<!-- A profile for running the benchmarks -->
<profile>
    <id>benchmark</id>
    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>${maven.exec.version}</version>
                <executions>
                    <execution>
                        <phase>test</phase>
                        <goals>
                            <goal>java</goal>
                        </goals>
                        <configuration>
                            <classpathScope>test</classpathScope>
                            <mainClass>javaslang.JmhRunner</mainClass>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</profile>
```

If it is started with `mvn test -P benchmark -X` we get this output (partial):

```bash
[DEBUG] Invoking : javaslang.JmhRunner.main()
[DEBUG] Plugin Dependencies will be excluded.
[DEBUG] Project Dependencies will be included.
[DEBUG] Collected project artifacts []
[DEBUG] Collected project classpath [/Users/daniel/git/temp/javaslang-benchmarkprofile/target/test-classes, /Users/daniel/git/temp/javaslang-benchmarkprofile/target/classes]
[DEBUG] Adding to classpath : file:/Users/daniel/git/temp/javaslang-benchmarkprofile/target/test-classes
[DEBUG] Adding to classpath : file:/Users/daniel/git/temp/javaslang-benchmarkprofile/target/classes
[DEBUG] joining on thread Thread[javaslang.JmhRunner.main(),5,javaslang.JmhRunner]
[WARNING] 
java.lang.ClassNotFoundException: javaslang.JmhRunner
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:281)
	at java.lang.Thread.run(Thread.java:745)
```

We see that the classpath isn't right.

**If we run `mvn test -pl javaslang-benchmark -P benchmark` it works.**

---

Our goal is to run it with `mvn test -P benchmark`.

Maybe the surefire plugin in the parent has to be configured to exclude javaslang-benchmark by default. The 'benchmark'  profile (in the parent pom) then configures surefire to include it. The javaslang-benchmark/pom.xml then includes the exec:java plugin config???